service.go

```diff
@@ -37,40 +37,143 @@ type Service struct {
 	sighupHandler func()
 }
 
-// New creates a new named service and allows for customization through functional options.
-func New(name, version string, oo ...OptionFunc) (*Service, error) {
+const fieldSetMsg = "Setting service property '%v' with '%v'"
+
+// Builder gathers all required properties to
+// construct a Patron service.
+type Builder struct {
+	errors        []error
+	name          string
+	version       string
+	cps           []Component
+	routes        []http.Route
+	middlewares   []http.MiddlewareFunc
+	acf           http.AliveCheckFunc
+	rcf           http.ReadyCheckFunc
+	termSig       chan os.Signal
+	sighupHandler func()
+}
+
+// NewBuilder initiates the Service builder chain.
+// The builder contains default values for Alive/Ready checks,
+// the SIGHUP handler and its version.
+func NewBuilder(name, version string) *Builder {
```

Comment:
Since it is the entrypoint for patron we can make simpler
```suggestion
func New(name, version string) *Builder {
```